### ISSUE TITLE
"overlayicon" - is this possible to change in a dialog in progress?

### ISSUE THREAD

**Issue by mm2270** [bartreardon/swiftDialog#115]
So, quick question. I was attempting to add an overlay icon to a dialog already in progress. I set up the initial dialog with an icon present, which is pushed to the background. The rest of the script continues and at some point I do:
```
echo "overlayicon:  /path/to/overlay_icon.png" >> /private/var/tmp/dialog.log
```
Or
```
echo "overlayicon:  SF=exclamationmark.circle.fill" >> /private/var/tmp/dialog.log
```

But neither of these have an effect. The original icon remains, but no overlay icon shows up.

If I add an overlay icon to the initial dialog, I am similarly unable to change it by echoing anything to the dialog.log control file.
Is it only possible to change the icon itself but not add or change an overlay icon like this?

**Comment by bartreardon** [comment]
Short answer, no - long answer...

not yet. Internally there are two ways dialog attributes are handled. the old way is static and the new way allows for dynamic updates. overlay icons are still handled the old way. The plan is to move everything to the dynamic. overlay icons are a little more complex than the main icon but I should have them done in an upcoming update.

**Comment by mm2270** [comment]
Ok, thanks for the update and explanation. I just wanted to make sure I wasn't missing something.

I'll just update the main icon for the purposes of what I was looking to do. It's actually pretty amazing that we can even do that for a running dialog window, so no complaints really. 😄

**Comment by bartreardon** [comment]
available in [1.11](https://github.com/bartreardon/swiftDialog/releases/tag/v1.11.0)
